packages/block-library/src/navigation-link/style.scss

```diff
@@ -205,6 +205,10 @@
 }
 
 // Default background and font color.
+// This includes blocks that have a background color,
+// as you might switch themes and the color specified is no longer registered.
+// But when one is explicitly chosen after the fact, it will override these rules.
+.wp-block-navigation.has-background,
```

Comment:
Right. Weird situation. The fact that we attach presets as classes make switching themes fragile. I hope this can be lessened by having "core" colors https://github.com/WordPress/gutenberg/issues/29568 but still there will be situations in which there may be preset theme colors.

I wonder if 29568 is a good opportunity to revisit how we do colors and whether presets can be inline colors (so we fix the theme switching issue).